src/datanode/src/heartbeat.rs

```diff
@@ -92,16 +99,31 @@ impl HeartbeatTask {
         let server_addr = self.server_addr.clone();
         let meta_client = self.meta_client.clone();
 
+        let catalog_manager_clone = self.catalog_manager.clone();
         let mut tx = Self::create_streams(&meta_client, running.clone()).await?;
         common_runtime::spawn_bg(async move {
             while running.load(Ordering::Acquire) {
+                let region_num = match region_number(catalog_manager_clone.clone()) {
```

Comment:
We'd better report a `region_num` value of `None` instead of just skip heartbeat reporting when fail to count regions on datanode.